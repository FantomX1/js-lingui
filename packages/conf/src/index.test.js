import { validate } from "jest-validate"
import {
  getConfig,
  replaceRootDir,
  catalogMigration,
  configValidation
} from "@lingui/conf"
import { mockConsole, getConsoleMockCalls } from "jest-mocks"

describe("@lingui/conf", function() {
  it("should return default config", function() {
    const config = getConfig()
    expect(config).toBeInstanceOf(Object)
    expect(config.sourceLocale).toBeDefined()
    expect(config.fallbackLocale).toBeDefined()
    expect(config.pseudoLocale).toBeDefined()
    expect(config.extractBabelOptions).toBeDefined()
    expect(config).toMatchSnapshot()

    // Deprecated and migrated configuration
    expect(config.localeDir).not.toBeDefined()
    expect(config.srcPathDirs).not.toBeDefined()
    expect(config.srcPathIgnorePatterns).not.toBeDefined()
  })

  it("should replace <rootDir>", function() {
    const config = replaceRootDir(
      {
        boolean: false,
        catalogs: {
          "<rootDir>/locale/{locale}/messages": [
            "<rootDir>/src",
            "!<rootDir>/ignored"
          ]
        }
      },
      "/Root"
    )

    expect(config.boolean).toEqual(false)
    expect(config.catalogs).toEqual({
      "/Root/locale/{locale}/messages": ["/Root/src", "!/Root/ignored"]
    })
  })

  describe("catalogMigration", function() {
    it("should show deprecation warning", function() {
      mockConsole(console => {
        validate(
          {
            localeDir: "./locale",
            srcPathDirs: ["./src"],
            srcPathIgnorePatterns: ["/node_modules/"]
          },
          configValidation
        )
        expect(getConsoleMockCalls(console.warn)).toMatchSnapshot()
      })
    })

    it("should provide default config", function() {
      const config = {}

      expect(catalogMigration(config)).toEqual({
        catalogs: {
          "locale/{locale}/messages": ["<rootDir>", "!node_modules/"]
        }
      })
    })

    it("should normalize string localeDir", function() {
      const config = {
        localeDir: "./locale"
      }

      expect(catalogMigration(config)).toEqual({
        catalogs: {
          "locale/{locale}/messages": ["<rootDir>", "!node_modules/"]
        }
      })
    })
    it("should migrate srcPathDirs and srcPathIgnorePatterns", function() {
      const config = {
        localeDir: "locale",
        srcPathDirs: ["src"],
        srcPathIgnorePatterns: ["src/node_modules/"]
      }

      expect(catalogMigration(config)).toEqual({
        catalogs: {
          "locale/{locale}/messages": ["src", "!src/node_modules/"]
        }
      })
    })
  })
})
