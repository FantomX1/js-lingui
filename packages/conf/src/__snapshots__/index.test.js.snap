// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@lingui/conf catalogMigration should show deprecation warning 1`] = `
● Deprecation Warning:

 Option localeDir is deprecated. Configure source paths using catalogs instead.
    
    @lingui/cli now treats your current configuration as:
    
    {
      "catalogs": {
  "locale/{locale}/messages": [
    "./src",
    "!/node_modules/"
  ]
}
    }
    
    Please update your configuration.
    

Documentation: https://lingui.js.org/ref/conf.html
● Deprecation Warning:

 Option srcPathDirs is deprecated. Configure source paths using catalogs instead.
    
    @lingui/cli now treats your current configuration as:
    
    {
      "catalogs": {
  "locale/{locale}/messages": [
    "./src",
    "!/node_modules/"
  ]
}
    }
    
    Please update your configuration.
    

Documentation: https://lingui.js.org/ref/conf.html
● Deprecation Warning:

 Option srcPathIgnorePatterns is deprecated. Configure excluded source paths using catalogs instead.
    
    @lingui/cli now treats your current configuration as:
    
    {
      "catalogs": {
  "locale/{locale}/messages": [
    "./src",
    "!/node_modules/"
  ]
}
    }
    
    Please update your configuration.
    

Documentation: https://lingui.js.org/ref/conf.html
`;

exports[`@lingui/conf should return default config 1`] = `
Object {
  catalogs: Object {
    locale/{locale}/messages: Array [
      /Users/tricoder/Projects/lingui/js-lingui,
      !node_modules/,
    ],
  },
  compileNamespace: cjs,
  extractBabelOptions: Object {
    plugins: Array [],
    presets: Array [],
  },
  fallbackLocale: ,
  format: lingui,
  locales: Array [],
  pseudoLocale: ,
  rootDir: /Users/tricoder/Projects/lingui/js-lingui,
  sourceLocale: ,
}
`;
